Add remember me cookie assertions

mdchaney opened the pull request:
Just adds the assertions mentioned in #87.

mdchaney committed:
Remove extraneous check for request.local?

Closes #88.

mdchaney committed:
Adds assertions for remember_me cookie.

Asserts cookie is http_only, secure, and same-site is "strict".
Closes #87.

stevepolitodesign commented on app/controllers/concerns/authentication.rb:
```diff
@@ -53,6 +53,6 @@ def user_signed_in?
 
   def store_location
-    session[:user_return_to] = request.original_url if request.get? && request.local?
+    session[:user_return_to] = request.original_url if request.get?
```
Was this meant to be part of #96?

mdchaney replied:
Yes.  Ignore this mess for now.

stevepolitodesign commented on test/controllers/sessions_controller_test.rb:
```diff
@@ -44,6 +44,12 @@ class SessionsControllerTest < ActionDispatch::IntegrationTest
     assert_not_nil current_user
     assert_not_nil cookies[:remember_token]
+
+    remember_me_cookie = cookies.get_cookie("remember_token")
+
+    assert remember_me_cookie.http_only?
+    assert remember_me_cookie.secure?
+    assert_equal "Strict", remember_me_cookie.to_h["SameSite"]
```
I'm surprised this passed, since it doesn't look like the implementation changed?

https://github.com/stevepolitodesign/rails-authentication-from-scratch/blob/b3e253fe2986d5672ba50f4ca23d4af038d5e8b1/app/controllers/concerns/authentication.rb#L37-L39

stevepolitodesign replied:
Ah, it did not pass.

@mdchaney would you be able to make the implementation change too, as outlined in [#53](https://github.com/stevepolitodesign/rails_mvp_authentication/issues/53)?

mdchaney replied:
Didn't pass.  I tested this in the wrong directory (long story).  I'm going to fix this.

mdchaney committed:
Adds security features to remember_token cookie.

1. Set to "secure" in production
2. Set to HttpOnly
3. SameSite set to strict.

Closes #87.

mdchaney committed:
Undoing prior accidental commit.

mdchaney commented:
Makeing a new pull request.
